### ISSUE TITLE
Expose functionality for renaming/moving notes through the CLI

### ISSUE THREAD

**Issue by benhsm** [dendronhq/dendron#1710]
## Please select if your request is either something new or an enhancement ##

- [ ] Enhancement of an existing Feature.
- [x] Request of a new feature.

## Please select the area your request applies to. (Multiple selections are possible) ##

- [ ] Onboard. Initial vault setup and import/export pods
- [ ] Create. Note creation, lookup, snippets and templates
- [ ] Retrieve. Backlinks, references, graph view
- [x] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Is your feature request related to a problem? Please describe ##

Currently, the [Dendron CLI](https://wiki.dendron.so/notes/23a1b942-99af-45c8-8116-4f4bb7dccd21/) does not have exposed functionality for renaming/moving existing notes.

This means that CLI users will have to resort to opening VSCode to change where a note is in a hierarchy / what its name is while preserving internal references to it.

## Describe the solution you'd like ##

I would want to be able to issue a command like:
```
dendron note (move|rename|mv) --wsRoot <dendron workspace> --query <note name> --newName <new name> [--fromVault <vault> --toVault <vault>]
```

Which would have the same effects as running [Dendron: rename note](https://wiki.dendron.so/notes/eea2b078-1acc-4071-a14e-18299fc28f47/#rename-note) or [Dendron: move note](https://wiki.dendron.so/notes/eea2b078-1acc-4071-a14e-18299fc28f47/#move-note) in the VSCode extension.

## Describe alternatives you've considered ##

I'm considering writing a shell script that would parse notes in a Dendron vault for references to a note to be renamed/moved and alter them accordingly along with the target note. This seems unnecessary when the logic for this is already in the engine, and it also wouldn't be as integrated with the rest of Dendron functionality.

## Additional context ##

I mainly want this for purposes of using Dendron through scripts and editors like Vim and Neovim without having to use VSCode. I'd like to eventually be able to do most of what is currently possible in the VSCode interface from the terminal, scripts and other editors, and I gather that this is already part of the long-term vision for Dendron. Searching around, I found one prior issue [#291](https://github.com/dendronhq/dendron/issues/291#issue-727622541) that this is possibly subsumed under.

The functionality for refactoring whole hierarchies is possibly related enough to be tackled alongside this.

I'm willing to muck around in the code myself to try and do this myself, but I would probably need some help and a lot of time since I don't have much experience, both in general and also with the Dendron codebase in particular.

Thanks for considering, and for planting and growing Dendron. 🌳

**Comment by kevinslin** [comment]
Thanks for the feature. Here is some documentation to help with the implementation. Feel free to post here or in the #dev channel if you run into any blockers or questions

- [Adding new subcommand to an existing command](https://wiki.dendron.so/notes/BiWzIKSDJBRBEhVQlZWpB.html#adding-new-subcommand-to-an-existing-command)
- [Note commands code](https://github.com/dendronhq/dendron/blob/cba633e4568601485e0cea1ab382e9dd3fbaa305/packages/dendron-cli/src/commands/notes.ts#L28)
- [Engine API](https://github.com/dendronhq/dendron-site/blob/dev/vault/pkg.dendron-engine.api.engine.md#L9:L9)

**Comment by benhsm** [comment]
Closing this issue because I think it's resolved by #2408
